### PR TITLE
feat: collect sync_optimizer and backwards pass timings [DET-5724]

### PR DESCRIPTION
**Pull request by stoksc**
## Description
This change just adds "accumulated timings" that get aggregated over a batch (since there may be many calls to ctx.backward() or ctx.step_optimizer()), and adds code to collect new metrics depending on this.

## Test Plan
- [x] add a test for edge cases with this code
- [x] other tests already exist

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: harness/determined/profiler.py [hunk 1]
```diff
@@ -145,10 +145,15 @@ def __init__(
...
         batch_idx: int,
         value: float,
+        accumulated: bool = False,
     ):
         super().__init__(timestamp, batch_idx, value)
         self.metric_type = metric_type
         self.metric_name = metric_name
+        self.accumulated = accumulated
+
+    def id(self) -> str:
```
Comment (armandmcqueen) [inline harness/determined/profiler.py:155]: This could be a property

**Review by armandmcqueen — APPROVED** [review]
LGTM
